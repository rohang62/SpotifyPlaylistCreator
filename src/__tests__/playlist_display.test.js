import React from "react";
import renderer from "react-test-renderer";
import PlaylistDisplay from "./../playlist_display";
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

/*jest.mock('axios', () => ({
    __esModule: true,
    post: jest.fn(() => Promise.resolve('test')),
    default: jest.fn(() => Promise.resolve('test')),
}));*/

test("renders loading screen", () => {
	let state = { name : '', username : '', minutes : 0 };
	const Tree = renderer.create(<PlaylistDisplay state = {state}/>).toJSON();
	expect(Tree).toMatchSnapshot();
});

function timeout(ms) {
	return new Promise(resolve => setTimeout(resolve, ms));
}

test("renders success screen", async () => {
	var mock = new MockAdapter(axios);
	let state = { name : '', username : '', minutes : 0 };
	mock.onPost('http://localhost:8000/create').reply(200, "test success");
	const asFragment = renderer.create(<PlaylistDisplay state = {state}/>);
	await timeout(3000);
	const tree = asFragment.toJSON();
	expect(tree).toMatchSnapshot();
});

test("renders error screen", async () => {
	var mock = new MockAdapter(axios);
	let state = { name : '', username : '', minutes : 0 };
	mock.onPost('http://localhost:8000/create').reply(400, "test error");
	const asFragment = renderer.create(<PlaylistDisplay state = {state}/>);
	await timeout(3000);
	const tree = asFragment.toJSON();
	expect(tree).toMatchSnapshot();
});
