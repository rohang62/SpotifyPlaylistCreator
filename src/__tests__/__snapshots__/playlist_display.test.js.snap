// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error screen 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="spotify_logo.svg"
    />
    <div
      className="Incorrect"
    >
      
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
       Go back to Login 
    </button>
  </header>
</div>
`;

exports[`renders loading screen 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="spotify_logo.svg"
    />
    Loading...
  </header>
</div>
`;

exports[`renders success screen 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="spotify_logo.svg"
    />
    <div
      className="Incorrect"
    >
      
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
       Go back to Login 
    </button>
  </header>
</div>
`;
